source/MaterialXRuntime/RtFileIo.h

```diff
@@ -65,6 +66,37 @@ class RtWriteOptions
     /// Filter function type used for filtering objects during write.
     /// If the filter returns false the object will not be written.
     WriteFilter writeFilter;
+
+    /// Enum that specifies how to generate material elements.
+    ///
+    /// NONE: don't generate material elements
+    ///
+    /// WRITE: generate material elements from surface shaders
+    ///
+    /// DELETE: delete source surface shaders (must be used with
+    /// WRITE)
+    ///
+    /// LOOK: generate a look for the material element (must be
+    /// used with WRITE)
+    ///
+    /// WRITE_DELETE: generate material elements from surface
+    /// shaders and delete the surface shaders
+    ///
+    /// WRITE_LOOKS: generate material elements for surface shaders
+    /// and write out looks
+    ///
+    /// WRITE_LOOKS_DELETE: generate material elements from
+    /// surface shaders, delete the surface shaders and write out
+    /// looks
+    enum MaterialElementType{ NONE               = 0,
```

Comment:
Maybe rename to something more descriptive. Since it's not used for material types but rather "material write operations", maybe rename to `MaterialWriteOp` or something like that.

Maybe also add a TODO comment that this is a temporary fix, since materials will be written as ordinary nodes when we get that change into the spec later.